proposals/testing/NNNN-image-attachments-in-swift-testing-windows.md

```diff
@@ -0,0 +1,391 @@
+  ///   - encodingQuality: The encoding quality to use when encoding images. For
+  ///     the lowest supported quality, pass `0.0`. For the highest supported
+  ///     quality, pass `1.0`.
+  ///
+  /// If the target image encoder does not support variable-quality encoding,
+  /// the value of the `encodingQuality` argument is ignored.
+  ///
+  /// If `clsid` does not represent an image encoder class supported by WIC, the
+  /// result is undefined. For a list of image encoder classes supported by WIC,
+  /// see the documentation for the [`IWICBitmapEncoder`](https://learn.microsoft.com/en-us/windows/win32/api/wincodec/nn-wincodec-iwicbitmapencoder)
+  /// class.
+  public init(encoderCLSID: CLSID, encodingQuality: Float = 1.0)
+}
+```
+
+For convenience, an initializer is provided that takes a path extension and
+tries to map it to the appropriate codec's `CLSID` value:
+
+```swift
+extension AttachableImageFormat {
+  /// Construct an instance of this type with the given path extension and
+  /// encoding quality.
+  ///
+  /// - Parameters:
+  ///   - pathExtension: A path extension corresponding to the image format to
+  ///     use when encoding images.
+  ///   - encodingQuality: The encoding quality to use when encoding images. For
+  ///     the lowest supported quality, pass `0.0`. For the highest supported
+  ///     quality, pass `1.0`.
+  ///
+  /// If the target image format does not support variable-quality encoding,
+  /// the value of the `encodingQuality` argument is ignored.
+  ///
+  /// If `pathExtension` does not correspond to a recognized image format, this
+  /// initializer returns `nil`:
+  ///
+  /// - On Apple platforms, the content type corresponding to `pathExtension`
+  ///   must conform to [`UTType.image`](https://developer.apple.com/documentation/uniformtypeidentifiers/uttype-swift.struct/image).
+  /// - On Windows, there must be a corresponding subclass of [`IWICBitmapEncoder`](https://learn.microsoft.com/en-us/windows/win32/api/wincodec/nn-wincodec-iwicbitmapencoder)
+  ///   registered with Windows Imaging Component.
+  public init?(pathExtension: String, encodingQuality: Float = 1.0)
+}
+```
+
+For consistency, `init(pathExtension:encodingQuality:)` is provided on Apple
+platforms too. (This is the only part of this proposal that affects platforms
+other than Windows.)
+
+### Example usage
+
+A developer may then easily attach an image to a test by calling
+`Attachment.record()` and passing the image of interest. For example, to attach
+an icon to a test as a PNG file:
+
+```swift
+import Testing
+import WinSDK
+
+@MainActor @Test func `attaching an icon`() throws {
+  let hIcon: HICON = ...
+  defer {
+    DestroyIcon(hIcon)
+  }
+  Attachment.record(hIcon, named: "my icon", as: .png)
+  // OR: Attachment.record(hIcon, named: "my icon.png")
+}
+```
+
+## Source compatibility
+
+This change is additive only.
+
+## Integration with supporting tools
+
+None needed.
```

Comment:
I think we could say something like (emphasis mine):

> No _additional_ tools integration is needed for this proposal. Common tools such as Swift package manager already include support for attachments, and the new kinds of image attachments proposed here will be covered under that.

("None needed" might suggest these attachments won't participate in any tools integration.)

Reply:
This section is identical to that in [ST-0014](https://github.com/swiftlang/swift-evolution/blob/main/proposals/testing/0014-image-attachments-in-swift-testing-apple-platforms.md#integration-with-supporting-tools).